types/wx-js-sdk-dt/index.d.ts

```diff
@@ -0,0 +1,568 @@
+         */
+        sourceType?: string[];
+
+        /**
+         * 返回选定照片的本地ID列表，localId可以作为img标签的src属性显示图片
+         */
+        success(res: { localIds: string[]}): void;
+    }
+
+    /**
+     * 拍照或从手机相册中选图接口
+     */
+    function chooseImage(config: ChooseImageConfig): void;
+
+    /**
+     * 显示照片预览用的配置对象
+     */
+    interface PreviewImageConfig {
+        /**
+         * 当前显示图片的 http 链接
+         */
+        current: string;
+        /**
+         * 需要预览的图片 http 链接列表
+         */
+        urls: string[];
+    }
+
+    /**
+     *  预览图片接口
+     * @param config
+     */
+    function previewImage(config: PreviewImageConfig): void;
+
+    /**
+     * localId:要上传的图片的本地 ID，由 chooseImage 接口获得
+     * isShowProgressTips: 默认为1，显示进度提示
+     * res: var serverId = res.serverId;  返回图片的服务器端 ID
+     */
+    interface UploadImageConfig {
+        localId: string;
+        isShowProgressTips?: number;
+        success(res: any): void;
+    }
+
+    function uploadImage(config: UploadImageConfig): void;
+
+    /**
+     * serverId: 需要下载的图片的服务器端ID，由 uploadImage 接口获得
+     * res: var localId = res.localId;  返回图片下载后的本地 ID
+     */
+    interface DownLoadImageConfig {
+        serverId: string;
+        isShowProgressTips?: number;
+        success?(res: any): void;
+    }
+
+    function downloadImage(config: DownLoadImageConfig): void;
+
+    /**
+     * locallId: 图片的 localID
+     */
+    interface getLocalImgDataConfig {
+        localId: string;
+        success(res: any): void;
+    }
+
+    /**
+     * 获取本地图片接口,此接口仅在 iOS WKWebview 下提供，用于兼容 iOS WKWebview 不支持 localId 直接显示图片的问题
+     */
+    function getLocalImgData(config: getLocalImgDataConfig): void;
+
+    /**
+     * 开始录音
+     */
+    function startRecord(): void;
+
+    /**
+     * 停止录音
+     * res: var localId = res.localId;
+     */
+    function stopRecord(success: (res: any) => void): void;
+
+    /**
+     * 录音时间超过一分钟没有停止的时候会执行 complete 回调
+     * @param complete :var localId = res.localId;
+     */
+    function onVoiceRecordEnd(complete: (res: any) => void): void;
+
+    /**
+     * 需要播放的音频的本地 ID，由 stopRecord 接口获得
+     * @param localId
+     */
+    function playVoice(localId: string): void;
+
+    /**
+     * 需要暂停的音频的本地 ID，由 stopRecord 接口获得
+     */
+    function pauseVoice(localId: string): void;
+
+    /**
+     * 需要停止的音频的本地 ID，由 stopRecord 接口获得
+     */
+    function stopVoice(localId: string): void;
+
+    /**
+     * res: var localId = res.localId;  返回音频的本地 ID
+     */
+    function onVoicePlayEnd(success: (res: any) => void): void;
+
+    interface UploadVoiceConfig {
+         localId: string;
+         isShowProgressTips?: number;
+         success(res: any): void;
+    }
+
+    /**
+     * 上传语音接口,上传语音有效期3天，可用微信多媒体接口下载语音到自己的服务器，
+     * 此处获得的 serverId 即 media_id，参考文档 .目前多媒体文件下载接口的频率限制为10000次/天，如需要调高频率，请登录微信公众平台，在开发 - 接口权限的列表中，申请提高临时上限。
+     * @param config res: var serverId = res.serverId; 返回音频的服务器端 ID
+     */
+    function uploadVoice(config: UploadVoiceConfig): void;
```

Comment:
The docs for `config` looks weird, is this the docs for `UploadVoiceConfig.success`?